File: boot-filepart/src/main/java/com/example/demo/PostRepository.java
package com.example.demo;

import org.springframework.data.domain.Pageable;
import org.springframework.data.r2dbc.repository.Query;
import org.springframework.data.r2dbc.repository.R2dbcRepository;
import org.springframework.data.repository.query.ReactiveQueryByExampleExecutor;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.UUID;

interface PostRepository extends R2dbcRepository<Post, UUID>, ReactiveQueryByExampleExecutor<Post> {

    public Flux<PostSummary> findByTitleLike(String title, Pageable pageable);
}

File: kotlin-co/src/main/kotlin/com/example/demo/repository/PostRepository.kt
package com.example.demo.repository

import com.example.demo.web.PostSummary
import com.example.demo.model.Post
import com.example.demo.model.Status
import kotlinx.coroutines.flow.Flow
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.*

interface PostRepository : CoroutineCrudRepository<Post, UUID> {
    fun findByTitleContains(title: String): Flow<PostSummary>
    fun findByStatus(status: Status): Flow<Post>
}